[Yaswitha Nalluri] iP

yastsc opened the pull request:
## Star ⭐️

Star is a chatbot that manages your life for you so that YOU can live up to your ⭐️ potential! It can do SO much:
- [ ] manage your deadlines
- [ ] manage your todos
- [ ] manage your events

And there's more! It is incredibly fun to use because its:
* text-based
* simple to learn
* fast to use

Start using Star NOW!
1. download it from [here]()
2. double-click it
3. add your tasks
4. sit back and watch it arrange everything for you 🤩
> Star, star, star, star, star, star - Megan Thee Stallion


You can also use this application to practise Java! Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(Star.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yastsc committed:
Rename files from Duke to Star

yastsc committed:
Add new greeting and exit

yastsc committed:
Echo commands entered by user and exit upon receiving command "bye"

yastsc committed:
Rename function from readScanner() to Echo()

yastsc committed:
Add function to store text entered by user and display back to user when requested

yastsc committed:
Add function to mark tasks as done or not done

yastsc committed:
Add 3 classes for ToDos, Deadlines and Events

yastsc committed:
Add a StarException class to deal with Star chatbot errors

yastsc committed:
Update the INPUT.txt, EXPECTED.txt, and runtest.sh to perform semi-automated testing of Star

yastsc committed:
Add functionality to delete tasks

yastsc committed:
Add function to save files in hard disk

yastsc committed:
Add date and time detection and formatting

yastsc committed:
Add date formatting for Event and Deadline tasks

yastsc committed:
Add new classes

yastsc committed:
Add relevant methods to Ui class

yastsc committed:
Add relevant methods to Storage class

yastsc committed:
Update Task and its subclasses to be compatible with the new classes

yastsc committed:
Add Command class and its subclasses

yastsc committed:
Add TaskList class

yastsc committed:
Complete methods for subclasses of Command

yastsc committed:
Add Parser class and the relevant methods

yastsc committed:
Change LocalDate to LocalDateTime to account for time

yastsc committed:
Organise classes into packages

yastsc committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

yastsc committed:
Set up Gradle to build and run Star

yastsc committed:
Adjust config in build.gradle

yastsc committed:
Add JUnit tests for Parser and ToDo

yastsc committed:
Create a JAR file for Star

yastsc committed:
Add JavaDoc comments

yastsc committed:
Change variable names to CamelCase to comply with Java Coding Standard

yastsc committed:
Add Find functionality to find tasks using keyword in description

yastsc committed:
Add checkStyle to detect coding standard violations

yastsc committed:
Add more JavaDoc comments to more classes

yastsc committed:
Change return type of methods in Ui class to String

yastsc committed:
Add a GUI to Star using JavaFX technology

yastsc committed:
Use Java Varargs for the dates for the Event task

yastsc committed:
Change class names to PascalCase to align with Java Coding Standard

EyuGongYi commented on src/main/java/star/command/Command.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Represents a Command that enables the execution of a set of actions.
+ */
+public class Command {
+
+    /**
+     * Executes the set of actions within.
+     *
+     * @param tasks   which is the task list on which the command is executed.
+     * @param ui      which is the UI that is used to interact with the user.
+     * @param storage which is the storage used to load and save tasks.
+     * @throws StarException which is an Exception specific to Star that is thrown when there is an error.
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws StarException {
+        return "";
+    }
```
Small thing to nitpick, but maybe you can change this to an abstract class? As execute dont really do anything and was thinking potentially down the line something might become an issue?

EyuGongYi commented on src/main/java/star/main/Storage.java:
```diff
@@ -0,0 +1,77 @@
+
+                Task task;
+                switch (inputs[0]) {
+                case "T":
+                    task = new ToDo(inputs[2]);
+                    break;
+                case "D":
+                    task = new Deadline(inputs[2], inputs[3]);
+                    break;
+                case "E":
+                    task = new Event(inputs[2], inputs[3]);
+                    break;
```
Nice use of switch , just thinking that you could add enum for the Task type in the case that you decide to change the icon for each of the task.

EyuGongYi commented on src/main/java/star/task/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+    @Override
+    public String getOutput() {
+        if (by == null) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy HH:mm");
```
Could this be extracted out? so that you dont have to redeclare them constantly and you dont have to search through the code to change each and everyone of it.

EyuGongYi commented on src/main/java/star/exception/StarException.java:
```diff
@@ -0,0 +1,48 @@
+ * Represents exceptions that are specific to the Star chatbot.
+ * Inherits from the Exception class.
+ */
+public class StarException extends Exception {
```
Like the way you do the different exceptions. Its quite an interesting way to handle it

EyuGongYi reviewed (commented):
Really love how the code is organised overall other than the things i commented. Good job on the code standards too

yastsc committed:
Add assertions to document important assumptions

There are several methods in which certain assumption(s) must hold in
order for the method to function as it is expected to.

Using the assert feature to document these assumptions allows us to
quickly verify and ensure that these assumptions hold.

yastsc committed:
Examine the code and refactor

In order to improve the code quality, refactoring the code was important.

Too long methods, arrowhead style code and complicated expressions were
refactored.

yastsc committed:
Use Stream to create a delay before Star's response

Streams can make code shorter and more functional.
Using a stream to create the delay, especially since it is not a very
essential part of the code

yastsc committed:
Merge pull request #1 from yastsc/branch-A-Assertions

Add assertions to document important assumptions

yastsc committed:
Add JavaDocs edits

yastsc committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/yastsc/ip into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/yastsc/ip:
  Examine the code and refactor

# Conflicts:
#	src/main/java/star/main/Parser.java

yastsc committed:
Merge pull request #2 from yastsc/branch-A-CodeQuality

Examine the code and refactor

yastsc committed:
Merge pull request #3 from yastsc/branch-A-Streams

Use Stream to create a delay before Star's response

yastsc committed:
Fix bugs in Parser

yastsc committed:
Add the Tagging function to the Task objects

Now, a tag can be added to the different Task objects. For now, it is only
one, but a future extension can be the addition of multiple tags.

Tags make it easier to categorise certain similar tasks. Another
future extension to the existing Find functionality can be to
find tasks tagged with a particular tag.

yastsc committed:
Fix some coding standard violations and add JavaDoc comments

yastsc committed:
Edit the Find functionality to make the variables' definition clearer

yastsc committed:
Shorten the parse method in the Parser class

yastsc committed:
Add more JavaDoc header comments

yastsc committed:
Add more through JUnit tests

yastsc committed:
Make the necessary edits to adhere to checkstyle

yastsc committed:
Handle error when Event time is not given

yastsc committed:
Update GUI to improve aesthetics

yastsc committed:
Add a User Guide for Star

yastsc committed:
Edit FindCommand to be clearer

yastsc committed:
Edit JavaDoc header comments

yastsc committed:
Fix formatting errors in User Guide

yastsc committed:
Fix bug to properly run JAR file on Windows/Linux

yastsc committed:
Fix bug to set correct mainClass

yastsc committed:
Fix bugs in build file related to mainClassName
